### PR TITLE
[s2024_road] Adding code that trains the model used traffic sign and traffic light detection module

### PR DESCRIPTION
**Pull request by ananaymathur**
This PR adds:

- traffic_sign_and_light_model_train.py to GEMstack/offboard/detection_learning/
- This is the code used to train the custom object detection model used in the traffic sign and traffic light detection module
- Code also runs the validation set

### CODE REVIEW

File: GEMstack/offboard/detection_learning/traffic_sign_and_light_model_train.py
Comment (krishauser) [inline GEMstack/offboard/detection_learning/traffic_sign_and_light_model_train.py:1]: This doesn’t seem to be saving anything or every printing anything. How will it be used?

**Review by krishauser — CHANGES_REQUESTED** [review]

File: GEMstack/offboard/detection_learning/traffic_sign_and_light_model_train.py
Comment (ananaymathur) [reply GEMstack/offboard/detection_learning/traffic_sign_and_light_model_train.py:1]: The model.train() function after it is done running automatically creates a .pt file which is the model.
The model.val() function prints to the console metrics from the validation run and also saves visual outputs like confusion matrices and graphs in a folder.

Screenshot 2024-05-06 at 5 16 02 PM

This is what the output looks after running the val() function.

confusion_matrix_normalized

This is the confusion matrix (one of the visual outputs) that is automatically generated and saved after the val() run.

Comment (krishauser) [reply GEMstack/offboard/detection_learning/traffic_sign_and_light_model_train.py:1]: I don’t see data.yaml, and where do you end up putting this trained model? A readme would be helpful.

**Review by krishauser — CHANGES_REQUESTED** [review]

**Comment by ananaymathur** [comment]
data.yaml is part of the dataset that the model is being trained on. I did not push the dataset to the repo.
I can add a readme explaining what the code does and also link the dataset in there.

**Comment by krishauser** [comment]
You should indeed add that documentation if your code is to be approved.

**Comment by ananaymathur** [comment]
Added a readme containing documentation

**Review by krishauser — APPROVED** [review]
